### ISSUE TITLE
PC should only use powers when needed

### ISSUE THREAD

**Issue by JonathanSafer** [jordansafer/screeps#70]
similarly to operate factory, the PC should not operate spawn, obs or extensions unless it is useful to the base. Otherwise we are just wasting Ops.

**Comment by JonathanSafer** [comment]
Basically operate obs should never get used right now since the obs never tries to long range scan

**Comment by JonathanSafer** [comment]
operate extensions and spawn are a little more difficult to determine when they should be used (maybe only during military action?)

**Comment by jordansafer** [comment]
there are functions canOperateObs, canOperateExtensions, canOperateSpawn (i think this is what they are called). So rn this issue is saying they should all return false? I think it's fair to change source/extension to detect if there's a hostile creep in the room or military flag placed

**Comment by jordansafer** [comment]
TLDR::
Add Util fns 
- underAttack(room)
- attackingEnemy(room)
- buildingNewSpawn(room)

1. Change operate spawn/extensions, so if any UtilFn returns true, we use them

**Comment by JonathanSafer** [comment]
Exactly my thought

**Comment by JonathanSafer** [comment]
Operate extensions and spawn are now activated by usage. Operate observer is disabled until an operated observer is even needed #12 . Operate controller is also activated by usage but drains Ops. As is this particular issue would seem to be resolved.
